test/validation/api/classification/odp_classification_test_pmr.c

```diff
@@ -369,13 +369,22 @@ static void _classification_test_pmr_term_tcp_dport(int num_pkt)
 	odp_pktio_close(pktio);
 }
 
-static void classification_test_pmr_term_tcp_sport(void)
+typedef enum match_t {
+	MATCH,
+	NO_MATCH
+} match_t;
+
+/*
+ * Test that PMR created using the given parameters matches or does not match
+ * given packet. The packet, that gets consumed, must have been created using
+ * create_packet() so that it contains the testing sequence number.
+ *
+ * Ethernet addresses of the packet will be overwritten.
+ */
+static void test_pmr(const odp_pmr_param_t *pmr_param, odp_packet_t pkt,
```

Comment:
This function looks good, but just a suggestion, Instead of creating and destroying PMR/COS sequence for every packet,
Can we take array of packets and result as input?
May be like this,
struct test_packet {
odp_packet_t pkt;
match_t match;
}test_packet;

Then the function will be something like this.
static void test_pmr(pmr_param, test_packet *tpkt[], num_packets)

Reply:
If there ever is a performance issue or other reason to avoid creating and destroying everything for each packet, I think it might be better to have a setup and cleanup function and have a simple single-packet interface for the actual test in between. But anyway, that more improvements can be done later if seen useful. I think this code ok for now and not much different from how things were (there are a bit more creation and destroying going on, but it should not really matter).